### PR TITLE
UI bug fixes reported

### PR DESCRIPTION
**Pull request by ajayadav09**
Refactored the center component and seperated the title component and the drop down component. It needed a local state to type in wothout re-rendering and added useEffect to update on the basis of that local state and change homepageTitle when content re-generates.

### CODE REVIEW

File: src/OnboardingSPA/steps/SiteGen/Editor/Header/center.js [hunk 1]
```diff
@@ -140,14 +123,89 @@ const StepEditorHeaderCenter = ( {
...
+		}
+	};
+
+	const onRenameItemSelect = () => {
+		enableInput( true );
+	};
+
+	const onRename = () => {
+		if ( typeof handleRename === 'function' ) {
+			handleRename( inputRef.current.value );
```
Comment (officiallygod) [inline src/OnboardingSPA/steps/SiteGen/Editor/Header/center.js:189]: Should we change the data in homepages if rename was selected and never edited? That does cause an unnecessary re-render. Also should we trim the data?

File: src/OnboardingSPA/steps/SiteGen/Preview/stylesheet.scss [hunk 2]
```diff
@@ -55,6 +55,8 @@
 		margin: 10px 10px 55px 10px;
 		flex-wrap: wrap;
 		max-width: 1440px;
+		align-items: center;
+		justify-content: center;
```
Comment (officiallygod) [inline src/OnboardingSPA/steps/SiteGen/Preview/stylesheet.scss:59]: Screenshot 2024-01-31 at 7 15 14 PM
Screenshot 2024-01-31 at 7 15 47 PM

Doesn't look right 👀

Comment (ajayadav09) [reply src/OnboardingSPA/steps/SiteGen/Preview/stylesheet.scss]: @officiallygod  Have updated the code with by adding width to the parent container. The width will be different for devices.

**Review by officiallygod — APPROVED** [review]
Looks good to me!

**Review by officiallygod — APPROVED** [review]

**Pull request merged**